src/layout/layout.rs

```diff
@@ -448,7 +448,7 @@ impl Layout {
     /// # }
     pub fn areas<const N: usize>(&self, area: Rect) -> [Rect; N] {
         let (areas, _) = self.split_with_spacers(area);
-        areas.to_vec().try_into().expect("invalid number of rects")
+        areas.as_ref().try_into().expect("invalid number of rects")
```

Comment:
I am curious what the types are before and after the `try_into` and which actual implementation is used here to look into it.
Sometimes into implementations use clone or stuff like that internally, so maybe this has the same outcome?

Reply:
AFAIU `as_ref` produces a `&[Rect]` slice, then `try_into` trivially casts it to `&[Rect; N]` and then copies that to the result array.

Reply:
Finding the actual implementations of (Try)Into stuff is horrible…

Looks like this still copies the `Rect`s, but that's way better than having a `Vec` on heap. :sparkles: